torchtune/modules/kv_cache.py

```diff
@@ -26,23 +26,43 @@ class KVCache(nn.Module):
 
     def __init__(
         self,
-        max_batch_size: int,
+        batch_size: int,
         max_seq_len: int,
         num_heads: int,
         head_dim: int,
         dtype: torch.dtype,
     ) -> None:
         super().__init__()
-        cache_shape = (max_batch_size, num_heads, max_seq_len, head_dim)
+        cache_shape = (batch_size, num_heads, max_seq_len, head_dim)
         self.register_buffer(
             "k_cache", torch.zeros(cache_shape, dtype=dtype), persistent=False
         )
         self.register_buffer(
             "v_cache", torch.zeros(cache_shape, dtype=dtype), persistent=False
         )
-        self.max_batch_size = max_batch_size
+        self.batch_size = batch_size
 
-    def update(self, input_pos, k_val, v_val) -> Tuple[Tensor, Tensor]:
+    def reset(self) -> None:
+        """Reset the cache to zero."""
+        self.k_cache.zero_()
+        self.v_cache.zero_()
+
+    def update(
+        self, input_pos: Tensor, k_val: Tensor, v_val: Tensor
+    ) -> Tuple[Tensor, Tensor]:
+        """Update KV cache and return the updated cache.
+
+        Args:
+            input_pos (Tensor):
```

Comment:
add shape comment (can also prob remove the code comment on L66 as it's now redundant)